tests/integration/schema_registry/test_json_serializers.py

```diff
@@ -0,0 +1,188 @@
+    """
+    Returns testProduct instance from it's dict format.
+
+    Args:
+        ctx (SerializationContext): Metadata pertaining to the serialization
+                operation.
+
+        product_dict (dict): testProduct in dict format.
+
+    Returns:
+        _TestProduct: product_obj instance.
+
+    """
+    return _TestProduct(product_dict['productId'],
+                        product_dict['productName'],
+                        product_dict['price'],
+                        product_dict['tags'],
+                        product_dict['dimensions'],
+                        product_dict['warehouseLocation'])
+
+
+def test_json_record_serialization(kafka_cluster, load_file):
+    """
+    Tests basic JsonSerializer and JsonDeserializer basic functionality
+
+    product.json from:
+        https://json-schema.org/learn/getting-started-step-by-step.html
+
+    Args:
+        kafka_cluster (KafkaClusterFixture): cluster fixture
+
+        load_file (callable(str)): JSON Schema file reader
+
+    """
+    topic = kafka_cluster.create_topic("serialization-json")
+    sr = kafka_cluster.schema_registry({'url': 'http://localhost:8081'})
+
+    schema_str = load_file("product.json")
+    value_serializer = JsonSerializer(sr, schema_str)
+    value_deserializer = JsonDeserializer(sr, schema_str)
+
+    producer = kafka_cluster.producer(value_serializer=value_serializer)
+
+    record = {"productId": 1,
+              "productName": "An ice sculpture",
+              "price": 12.50,
+              "tags": ["cold", "ice"],
+              "dimensions": {
+                  "length": 7.0,
+                  "width": 12.0,
+                  "height": 9.5
+              },
+              "warehouseLocation": {
+                  "latitude": -78.75,
+                  "longitude": 20.4
+              }}
+
+    producer.produce(topic, value=record, partition=0)
+    producer.flush()
+
+    consumer = kafka_cluster.consumer(value_deserializer=value_deserializer)
+    consumer.assign([TopicPartition(topic, 0)])
+
+    msg = consumer.poll()
+    actual = msg.value()
+
+    assert all([actual[k] == v for k, v in record.items()])
+
+
+def test_json_record_serialization_incompatible(kafka_cluster, load_file):
+    """
+    Tests Serializer validation functionality.
+
+    product.json from:
+        https://json-schema.org/learn/getting-started-step-by-step.html
+
+    Args:
+        kafka_cluster (KafkaClusterFixture): cluster fixture
+
+        load_file (callable(str)): JSON Schema file reader
+
+    """
+    topic = kafka_cluster.create_topic("serialization-json")
+    sr = kafka_cluster.schema_registry({'url': 'http://localhost:8081'})
+
+    schema_str = load_file("product.json")
+    value_serializer = JsonSerializer(sr, schema_str)
+    producer = kafka_cluster.producer(value_serializer=value_serializer)
+
+    record = {"contractorId": 1,
+              "contractorName": "David Davidson",
+              "contractRate": 1250,
```

Comment:
hot daymn! I need to learn masonry

Reply:
David Davidson nis really good but sometimes shows up when all you wanted was an ice swan.